### PR TITLE
Cj build1

### PR DESCRIPTION
**Pull request by 8v060htwyc**

### CODE REVIEW

File: build/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+RUN tar xvfz nasm-2.11.06.tar.gz 
+WORKDIR /tmp/nasm-2.11.06/
+RUN ./configure
+RUN make
+RUN make install 
+
+WORKDIR /tmp
+RUN git clone https://github.com/cisco/openh264.git
+WORKDIR /tmp/openh264
+RUN git checkout v1.1
```
Comment (sijchen) [inline build/Dockerfile]: intentionally checkout v1.1 here -- is this needed to be updated with newer versions or not?

**Comment by shihuade** [comment]
1. what about cent-OS, apt-get should be replace with yum command, is it possible that automatically change  command like apt-get to yum?

2. for below command, the update may take such a long time to update the entire packages, 
       RUN apt-get update
       RUN apt-get upgrade -y
       RUN apt-get install -y \

**Comment by shihuade** [comment]
for below RUN instruction, it seems that it is not follow the best practices guid  which  you can refer to https://docs.docker.com/articles/dockerfile_best-practices/

+RUN apt-get update
+RUN apt-get upgrade -y
+RUN apt-get install -y bison flex g++ gcc git libgmp3-dev libmpc-dev libmpfr-dev libz-dev make wget
+

**Comment by shihuade** [comment]
1. There are some errors when I used an totally new Ubuntu machine with  version(14.04 or 15.04).
  it seems that blocked by below instruction:
   RUN apt-get install -y bison flex g++ gcc git libgmp3-dev libmpc-dev libmpfr-dev libz-dev make wget
 but when only used apt-get outside the dockerfile, everything is OK.

***********************************************************************************
The following packages have unmet dependencies:
 g++ : Depends: cpp (>= 4:4.8.2-1ubuntu6) but it is not going to be installed
       Depends: g++-4.8 (>= 4.8.2-5~) but it is not going to be installed
       Depends: gcc-4.8 (>= 4.8.2-5~) but it is not going to be installed
 gcc : Depends: cpp (>= 4:4.8.2-1ubuntu6) but it is not going to be installed
       Depends: gcc-4.8 (>= 4.8.2-5~) but it is not going to be installed
       Recommends: libc6-dev but it is not going to be installed or
                   libc-dev
 zlib1g-dev : Depends: libc6-dev but it is not going to be installed or
                       libc-dev
E: Unable to correct problems, you have held broken packages.
***********************************************************************************

**Comment by HaiboZhu** [comment]
It is too complex for our codec, thank you for your contribution

**Pull request closed without merging**